core/provider/digitalocean/ssh.go

```diff
@@ -41,41 +43,71 @@ func makeSSHKeyPair() (string, string, string, error) {
 }
 
 func getUserIPs(ctx context.Context) (ips []string, err error) {
-	res, err := http.Get("https://ifconfig.io")
```

Comment:
ideally here, we try both ifconfig.io and ifconfig.co (different ASNs, so they'll get routed through different paths) and try both IPv4 and IPv6. we should also add the option in the provider to append a custom IP